### ISSUE TITLE
[Bug] - Enhanced Idle seems to sometimes rely on *current* window instead of MAS window.

### ISSUE THREAD

**Issue by TheElementalOfDestruction** [Monika-After-Story/MonikaModDev#7050]
I opened a window that was taking up half of the screen over the MAS window and saw that if I went to the side of that window (but still had my cursor over the MAS window) that Monika's eyes would look to that side. I then tried to reproduce this and would only be able to get it to happen very occasionally. 

I've provided a sample image of how I had my windows setup and where my mouse was. In my case, she was looking to the right side of the screen when my cursor was at the same position as the large red dot that I put in the image.
![image](https://user-images.githubusercontent.com/24580325/109426268-8c91e800-79a1-11eb-891e-223a49639e64.png)

**Comment by TheElementalOfDestruction** [comment]
I managed to get it to happen again, but I don't know why it happens. I actually managed to take a picture using my phone so you could see my cursor as direct proof that this is happening: 
![image0](https://user-images.githubusercontent.com/24580325/110237439-f6a21400-7ef0-11eb-9a71-823cdb9679a4.jpg)

**Comment by TheElementalOfDestruction** [comment]
I don't think this is related to any of the submods I have, but *just* in case here is a list of the ones I have installed:
* Auto Atmos Change
* Better Loading
* Custom Room Den
* Custom Room Furnished Spaceroom V1
* Custom Room Furnished Spaceroom V2
* Custom Room Furnished Spaceroom V3
* Custom Room Kitchen
* Submod Updater Plugin

**Comment by Booplicate** [comment]
What was the name of that second window?

**Comment by TheElementalOfDestruction** [comment]
Every time it's happened so far it's been Windows Explorer (this time it was the Monika after story folder, but I *think* is was a different one the last time)

**Comment by TheElementalOfDestruction** [comment]
And I literally named the folder "Monika After Story" and it might be looking by window name. I'll double check if it happens with the same folder when I get back to my computer.

**Comment by Booplicate** [comment]
I found the issue, will be fixed in next release. Thanks for the report!

**Comment by ThePotatoGuy** [comment]
[closing as fixed]

**Comment by TheElementalOfDestruction** [comment]
This issue was supposed to be fixed in 0.12.2, correct?

**Comment by TheElementalOfDestruction** [comment]
Can confirm 2 versions later this is *not* fixed. Found out how to easily replicate it and it still happens.

**Comment by TheElementalOfDestruction** [comment]
Easy replication method on windows 10: Name a folder "Monika After Story", open that folder in Windows Explorer. Move that to the side of the actual MAS window and when she goes to start doing the enhanced idle it will use that window instead.

**Comment by multimokia** [comment]
This issue is a little tricky to fix given what _little_ information we get out of explorer there. The code to find the MAS window is as follows:

```python
    def __getMASWindowHWND():
        """
        Gets the hWnd of the MAS window

        NOTE: Windows ONLY

        OUT:
            int - represents the hWnd of the MAS window
        """
        #Verify we can actually do this before doing anything
        if not store.mas_windowreacts.can_do_windowreacts:
            return None

        def checkMASWindow(hwnd, lParam):
            """
            Internal function to identify the MAS window. Raises an exception when found to allow the main func to return
            """
            if renpy.config.window_title == win32gui.GetWindowText(hwnd):
                raise MASWindowFoundException(hwnd)

        try:
            win32gui.EnumWindows(checkMASWindow, None)

        except MASWindowFoundException as e:
            return e.hwnd
```

Where we throw an Exception to break the loop and return the information from the inner func.

But this checks the known window title of MAS against the one we got from the window associated with the `hwnd`. Additionally I've noticed the active window gets *priority* from this as well (it occurs earlier in the list), hence why you're seeing that behaviour.

Unfortunately, Explorer does _not_ provide the process name in the window title. As such it is more or less _impossible_ to distinguish between MAS and a folder with the same name.

This may not really be a fixable problem.

**Comment by Booplicate** [comment]
> Easy replication method on windows 10: Name a folder "Monika After Story", open that folder in Windows Explorer. Move that to the side of the actual MAS window and when she goes to start doing the enhanced idle it will use that window instead.

What version of Windows do you use? Windows explorer shows the full path in its title, naming a folder `Monika After Story` doesn't trigger this. Most programs have other things (prefixes/suffixes) in their titles, so overall this is unlikely to be a problem.

**Comment by TheElementalOfDestruction** [comment]
Windows 10 Home version 10.0.19042 Build 19042 is what I get when I ask for my system information. I can also confirm I can replicate this by using the "title" command in CMD.

**Comment by TheElementalOfDestruction** [comment]
One thing that could be done would be to check the process id based on the hwnd, but I think you need access to win32process which doesn't seem to be available with the currently provided python library. Specifically I would have used `GetWindowThreadProcessId` and then tried to compare against the process ID MAS is using. Although that also assumes that the process ID doesn't change at all (does it?).

But again, I don't see the function in the currently provided libraries.

**Comment by Booplicate** [comment]
We'll just add a special suffix to the title.
